io/disk/iozone.py

```diff
@@ -0,0 +1,649 @@
+                    w_count = int(match.group(1))
+                    section = self.__get_section_name(match.group(2))
+
+                    # Output the appropriate keyval pair
+                    key_name = '%s-%d-kids' % (section, w_count)
+                    keylist[key_name] = match.group(3)
+                    continue
+
+                # Check for any other interesting lines
+                if '=' in line:
+                    # Is it something we recognize? First check for parent.
+                    match = parent_regexp.search(line)
+                    if match:
+                        # The section name and the worker count better match
+                        p_count = int(match.group(1))
+                        p_secnt = self.__get_section_name(match.group(2))
+                        if p_secnt != section or p_count != w_count:
+                            continue
+
+                        # Set the base name for the keyval
+                        basekey = 'parent'
+                    else:
+                        # Check for the various 'throughput' values
+                        if line[3:26] == ' throughput per thread ':
+                            basekey = line[0:3]
+                            match_x = KBsec_regexp
+                        else:
+                            # The only other thing we expect is 'Min xfer'
+                            if not line.startswith('Min xfer '):
+                                continue
+                            basekey = 'MinXfer'
+                            match_x = KBval_regexp
+
+                        match = match_x.search(line)
+                        if match:
+                            result = match.group(1)
+                            key_name = "%s-%d-%s" % (section, w_count, basekey)
+                            keylist[key_name] = result
+        output_path = os.path.join(self.outputdir, "perf.json")
+        json.dump(keylist, open(output_path, "w"), indent=1)
+
+    def test(self):
+
+        dir = self.params.get('dir', default=None)
+        args = self.params.get('args', default=None)
+
+        if not dir:
+            dir = self.base_dir
+        os.chdir(dir)
+
+        if not args:
+            args = '-a'
+
+        cmd = os.path.join(self.srcdir, 'src', 'current', 'iozone')
+        self.results = process.system_output('%s %s' % (cmd, args))
+        self.auto_mode = ("-a" in args)
+        iteration = 1
+        self.results_path = os.path.join(self.outputdir,
+                                         'raw_output_%s' % iteration)
+        self.analysisdir = os.path.join(self.outputdir,
+                                        'analysis_%s' % iteration)
+        f = open(self.results_path, 'w')
+        try:
+            f.write(self.results)
+        finally:
+            f.close()
+
+        self.generate_keyval()
+
+        if self.auto_mode:
+            a = IOzoneAnalyzer(list_files=[self.results_path],
```

Comment:
Actually you might skip this one as it's an improvement and not port. Or if you're interested, you can include it as a separate commit.